x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/CcrDisabledIT.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ccr;
+
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.xpack.core.XPackSettings;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+public class CcrDisabledIT extends ESIntegTestCase {
+
+    public void testClusterCanStartWithCcrInstalledButNotEnabled() throws Exception {
+        ensureGreen();
```

Comment:
I think that the problem that is fixed here only occurs when an index is created, so let's create an index in this test. Perhaps also invoke the `XPackClient.prepareInfo()` API to check whether the plugin is disabled.

Reply:
I mean I specifically tried and the test failed without the my fix in the `Ccr.java` class. You can see it happens when the add listener method is called:

```
    public void addIndexEventListener(IndexEventListener listener) {
        ensureNotFrozen();
        if (listener == null) {
            throw new IllegalArgumentException("listener must not be null");
        }
        if (indexEventListeners.contains(listener)) {
            throw new IllegalArgumentException("listener already added");
        }

        this.indexEventListeners.add(listener);
    }
```

I will add the `prepareInfo` check.

Reply:
Also - it looks like CCR has not implemented the `XPackFeatureSet` functionality? When I search for references, it is not referenced in the ccr package.

Reply:
There is a PR open for this.